### DISCUSSION TITLE
Ridiculous startup times.

### DISCUSSION THREAD

**Discussion by praveenc1** [AnswerDotAI/fasthtml#402] [General]
I have startup times that are extremely painful. Before I logged an issue, I thought I'll first do some preliminary checks with support from folks here.

I run `python test.py` and just wait about a minute or two before the server startup and serves the page. This is on a MacBook Pro M1. I suspected iCloud maybe moving files to iCloud and it's having to download from iCloud. But even when I force a download this delay exists. What can I do to troubleshoot this? Please help!

test.py
-------
```
from fasthtml.common import *
app, rt = fast_app()

@app.get("/")
async def root():
    return {"message": "Hello World"}
```

**Comment by johnowhitaker** [comment]
How are you running the app? Running with uvicorn or adding `serve()` both result in a running app in a second or two for me. You could see if its fasthtml-specific by making a file with 

```python
async def app(scope, receive, send):
    assert scope['type'] == 'http'

    await send({
        'type': 'http.response.start',
        'status': 200,
        'headers': [
            [b'content-type', b'text/plain'],
        ],
    })
    await send({
        'type': 'http.response.body',
        'body': b'Hello, world!',
    })
```

And running with `uvicorn main:app --port 8001` (assuming main.py) to see if that starts a uvicorn server quickly or with a delay.

**Reply by praveenc1** [reply]
@johnowhitaker Same thing with that sample too. Doesn't look like it has anything to do with FastHTML. Frustrating thing is, its sporadic.

**Comment by dhirschfeld** [comment]
If on Windows, Defender can inject itself to scan binaries before allowing the app to start.
